doc/sphinx/proofs/writing-proofs/proof-mode.rst

```diff
@@ -144,6 +144,8 @@ list of assertion commands is given in :ref:`Assertions`. The command
    .. seealso:: :cmd:`Proof with`
 
 .. cmd:: Proof using @section_var_expr {? with @ltac_expr }
+   :undocumented:
+.. cmdv:: Proof with @ltac_expr {? using @section_var_expr }
```

Comment:
That's very far from what I was suggesting. I was suggesting to have only:

```rst
.. cmd:: Proof using @section_var_expr
```

here and

```rst
.. cmd:: Proof with @ltac_expr {? using @section_var_expr }
         Proof using @section_var_expr with @ltac_expr
```

in the `auto.rst` file.

This would correspond to the various forms available in `fullGrammar`:

https://github.com/coq/coq/blob/28c8d0f6fd7fc07c51edaa9b2da020c70f03a79e/doc/tools/docgram/fullGrammar#L483

https://github.com/coq/coq/blob/28c8d0f6fd7fc07c51edaa9b2da020c70f03a79e/doc/tools/docgram/fullGrammar#L607-L608

If that would not be supported by `doc_grammar` because you can only have one place where `Proof using` is described, then it's better to backtrack to the previous presentation.

Reply:
Doing as you suggest would require making one of the `Proof using`s a `:cmdv:` (I think that's OK), but there's no way for a single index entry to point to 2 places.  You'd have to use a different name for one of them.  And I don't like splitting up the description of `Proof using`--it breaks the pattern for cmds/tactics.  I'll revert this change.

> This would correspond to the various forms available in `fullGrammar`:

Not very persuasive given how freely we've deviated from `fullGrammar`, with 2800 lines of edits in `common.edit_mlg`.